lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,109 @@
+module BeakerPuppet
+  module Install
+    module Puppet5
+
+      # grab build json from the builds server
+      #
+      # @param [String] project_name Name of the project
+      # @param [String] project_sha SHA ref of the build for the project
+      #
+      # @return [String, String] URL to the build for the SHA of the project,
+      #   build json parsed into a ruby hash
+      def fetch_build_details(project_name, project_sha)
+        buildserver_url = 'http://builds.delivery.puppetlabs.net'
+        buildserver_projectsha_url = "#{buildserver_url}/#{project_name}/#{project_sha}"
+        file_folder = "#{buildserver_projectsha_url}/artifacts"
+        filename = "#{project_sha}.yaml"
+        dst_folder = Dir.pwd
+
+        # fetch & parse file
+        file_local_path = fetch_http_file(file_folder, filename, dst_folder)
+        file_hash = YAML.load_file(file_local_path)
+
+        # getting the relevant data from the file
+        logger.perf_output("YAML HASH BELOW:")
+        logger.perf_output(file_hash)
+        # pp file_hash
+        logger.perf_output("PLATFORM_DATA BELOW:")
+        logger.perf_output(file_hash[:platform_data])
+        return buildserver_projectsha_url, file_hash
+      end
+
+      # get the build artifact URL for a host
+      #
+      # @param [Host] host Host to get artifact URL for
+      # @param [Hash] build_details Details of the build in a hash
+      # @param [String] build_url URL to the build
+      #
+      # @return [String] URL to the build artifact
+      def hosts_artifact_url(host, build_details, build_url)
+        platform_key = host[:platform]
+        variant, version, arch, _ = platform_key.to_array
+        # step: map platform strings into platform_key's for the hash
+        if variant == 'debian' && arch == 'x86_64'
+          platform_key.sub!('x86_64', 'amd64')
+        elsif variant == 'fedora'
+          platform_key.sub!(version, "f#{version}")
+        end
```

Comment:
@joshcooper @kevpl so if I'm understanding correctly, you would like the fedora platform tags to always be of the form fedora-<ver>-<arch> and to always use x86_64 instead of a mix of that and amd64? The former should be easy; I will look into what it would take for the latter.